src/nacl/types.py

```diff
@@ -0,0 +1,13 @@
+import sys
+
+if sys.version_info >= (3, 8):
+    from typing import SupportsBytes
+else:
+    from typing_extensions import Protocol
+
+    class SupportsBytes(Protocol):
```

Comment:
What would be the disadvantage of just defining the types where you use `SupportsBytes` as just `bytes`? Clearly that makes it so types that can be coerced into bytes won't be detected, but is that a common case?

Reply:
Looking at that again, I don't see any indication that `SupportsBytes` in 3.8+ only. It appears to be available in 3.6, so this should be an unconditional import.

Reply:
Which would also mean that there's no need to have this file, just import from `typing`.

Reply:
> What would be the disadvantage of just defining the types where you use `SupportsBytes` as just `bytes`? Clearly that makes it so types that can be coerced into bytes won't be detected, but is that a common case?

No huge disadvantage; I just wanted to give mypy as complete a picture as possible, and avoid a `# type: ignore[foo]` unless necessary.

I only want to use this for `StringFixer` and `Encodable` (even in the big typedump of #692.) Not massively common.

Given that SupportsBytes is in 3.6 and up, I can just import directly as @Dreamsorcerer says and there's no need for any of this faff. How does that sound?

> Looking at that again, I don't see any indication that `SupportsBytes` in 3.8+ only. It appears to be available in 3.6, so this should be an unconditional import.

Thanks for spotting this. (I assumed that SupportsBytes wasn't available until 3.8 because that's when `Protocol` was made available; but it looks like [3.6 was using `_Protocol` internally](https://github.com/python/cpython/blob/2c56c97f015a7ea81719615ddcf3c745fba5b4f3/Lib/typing.py#L1777-L1782).

Reply:
I am happy with SupportsBytes since it supports 3.6+ without typing_extensions, yes

Reply:
Also, this was broken on 3.7 and below: `typing_extensions` wasn't marked as a requirement in `setup.py`.